openssl/src/symm.rs

```diff
@@ -135,8 +135,7 @@ impl Crypter {
     ///
     /// # Panics
     ///
-    /// Panics if an IV is required by the cipher but not provided, or if the
-    /// IV's length does not match the expected length (see `Cipher::iv_len`).
+    /// Panics if an IV is required by the cipher but not provided.
```

Comment:
Can this constraint just be encoded via the type system? Maybe have `new` and `new_with_iv` where `new` takes a subset of `Cipher` that doesn't require an initialization vector?